src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -423,18 +423,13 @@ public bool MoveNext()
 
                         _reduceKeyProcessor.Reset();
 
-                        var propertiesInOrder = accessor.PropertiesInOrder;
-                        int properties = propertiesInOrder.Count;
-
-                        for (int i = 0; i < properties; i++)
+                        foreach (var property in accessor.GetPropertiesInOrder(output))
```

Comment:
I remember that was changed intentionally to use `for` instead of `foreach` by @redknightlois due to perf optimization. Probably not critical but worth to keep it in mind

Reply:
At the time it was a big offender, mostly because of very bad assembler generated and a very hot method.